packages/jupyter-ai/jupyter_ai/chat_handlers/base.py

```diff
@@ -471,3 +479,128 @@ def send_help_message(self, human_msg: Optional[HumanChatMessage] = None) -> Non
         )
 
         self.broadcast_message(help_message)
+
+    def _start_stream(self, human_msg: HumanChatMessage) -> str:
+        """
+        Sends an `agent-stream` message to indicate the start of a response
+        stream. Returns the ID of the message, denoted as the `stream_id`.
+        """
+        stream_id = uuid4().hex
+        stream_msg = AgentStreamMessage(
+            id=stream_id,
+            time=time.time(),
+            body="",
+            reply_to=human_msg.id,
+            persona=self.persona,
+            complete=False,
+        )
+
+        self.broadcast_message(stream_msg)
+        return stream_id
+
+    def _send_stream_chunk(
+        self,
+        stream_id: str,
+        content: str,
+        complete: bool = False,
+        metadata: Dict[str, Any] = {},
```

Comment:
Just highlighting a pre-existing potential issue here (https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil)

Reply:
Thanks for calling this out! I'll update the code & keep this mind in the future.

It would be better if there was some automated method of catching this for other devs. Do you know of any linter that can flag mutable default arguments?